packages/network-controller/tests/fake-provider-engine.ts

```diff
@@ -1,18 +1,20 @@
 import { inspect, isDeepStrictEqual } from 'util';
 import EventEmitter from 'events';
-import type {
-  ProviderEngine,
-  RpcPayload,
-  RpcResponse,
-} from 'web3-provider-engine';
-
+import {
+  JSONRPCRequest,
+  JSONRPCResponse,
+} from '@json-rpc-specification/meta-schema';
+import { SafeEventEmitterProvider } from '@metamask/eth-json-rpc-provider/dist/safe-event-emitter-provider';
+import { JsonRpcEngine } from 'json-rpc-engine';
 // Store this in case it gets stubbed later
 const originalSetTimeout = global.setTimeout;
 
 /**
  * The payload that `sendAsync` takes.
  */
-type SendAsyncPayload<P> = RpcPayload<P> | RpcPayload<P>[];
+interface SendAsyncPayload<P> extends Omit<JSONRPCRequest, 'params'> {
```

Comment:
We should be able to get this from `json-rpc-engine` now instead of filling this in.

Reply:
Hmm, sorry, I guess `json-rpc-engine` uses `JsonRpcRequest`, but it looks like we've moved it to `@metamask/utils` (proof [here](https://github.com/MetaMask/utils/blob/f424d21ff042f96852da60ab419949aab6454388/src/json.ts#L218)).

Reply:
Ah there is the JsonRpcRequest type from `@metamask/utils`